### DISCUSSION TITLE
More control over notifications

### DISCUSSION THREAD

**Discussion by lehresman** [voxeltycoon/issues#400] [UI&UX]
There are too many notifications that I don't care about.  So many that I've just started ignoring them.

One of my cities got starved for resources because of a train deadlock, and I didn't notice because I had tuned out the notifications.

Suggestions for improvement:
* Only notify about industries/businesses that you are supplying
* Have a way to watch an industry/business that I may care about in the future

**Comment by juffanoodle** [comment]
Maybe split the notifications out into categories, instead of having them all in one list? 

* Lost vehicles
* Starving businesses
* New businesses
* Increased capacity
* Research
* ....

**Comment by victorleblais** [comment]
I'm going to make a mix of both suggestions:

Have a "global" notification settings where you can toggle on/off notifications about:

- Lost vehicles
- New businesses
- Increased capacity
- Research
- Resource depletion
- ...

**and** also have a new tab on detail windows of cities where you can:

- See a list of all businesses in that city and you can select for each:
    - No notification at all
    - Minimal notification (only when the business is about to close down)
    - All notifications (increase/decrease of demand)
- and also three buttons "Set for all" that applies the same setting (none/minimal/all) to all business at once

Businesses would also have a tab to set their own setting, which would sync with the tab of the city.

That way you can entirely mute cities that you don't care about, and also mute "variations" in demand of cities that you already service, but don't care about the details of it (you just want to know if businesses are going to close); meanwhile you still have the possibility to be up to date of all changes on a few businesses that you're actively trying to grow.

**Comment by int3ger** [comment]
Dealing with notifications could also be made more friendly by removing their ability to forcefully move your current map view and instead popup the dialog for the given city, business, vehicle etc whereby we can CHOOSE to travel to that destination via another click or just take care of the issue via the dialog box. Provide a setting so this could be enabled/disabled as desired.

**Comment by victorleblais** [comment]
You can now have a setting to ignore supplying notifications for buildings you don't service (or all buildings):
![image](https://user-images.githubusercontent.com/4365075/115538604-5c2d4280-a29c-11eb-9e7e-ade6fde2f231.png)

You can also ignore a vehicle's notifications
![image](https://user-images.githubusercontent.com/4365075/115538704-736c3000-a29c-11eb-8145-a0ae8134e527.png)

I'm renaming the issue to reflect the desire to have even more granular control on this

**Reply by RockSoftly** [reply]
Consider adding more options that allow us to control which notifications we see. For example, it would be nice if notifications related to a particular tier of product could be ignored (no notice added to the list). To reduce traffic, I often stop delivering the base resources to consumers – such as ore, coal, sand, stone and wood. I get a lot of notifications regarding new and existing consumers of these products. This is an annoyance and creates more trash to scan through when I look at my notifications. I would also like to be able to ignore notifications regarding products that I do not currently make or mine.

**Comment by SRS-CHI** [comment]
I've found that with passengers this has become newly important to me. Buses will occasionally run empty and when they stop while empty while set to "transfer" they generate the message "Car #x has nothing to unload."  This crowds out important notifications and it doesn't seem I can mute these without also muting those.
